content/articles/building-a-machine-learning-classification-model-with-pycaret/index.md

```diff
@@ -0,0 +1,191 @@
+
+```python
+    !pip install pycaret pandas shap 
+```
+
+If you're installing these dependencies using your local jupyter notebook, no need to put the exclamation `!` before the `pip` command.
+
+#### Importing dependencies
+Let's now import these dependencies into our Google Colab:
+
+```python
+    import pandas as pd
+    from pycaret.classification import *
+```
+
+### Loading custom dataset from Kaggle using Pandas
+Let's go ahead and download the [Default of Credit Card Clients Dataset](https://www.kaggle.com/uciml/default-of-credit-card-clients-dataset) dataset from Kaggle. Grab the downloaded dataset from the Downloads folder on your computer and copy it into the Google Colab folder that you're working on.
+
+We can then load this dataset in our Colab using the pandas library:
+
+```python
+    df = pd.read_csv(UCI_Credit_Card.csv)
+```
+To view it, let's type in the following:
+
+```python
+    df.head()
+```
+
+Alternatively, there exists a built-in PyCaret's data repository. Using the `get_data()` function, you can directly load the data into your Colab. But, this option requires you to have an internet connection.  
+
+```python
+    from pycaret.datasets import get_data
+    credit_dataset = get_data('credit')
+```
+
+![Loading the Default of Credit Card Clients Dataset](/engineering-education/building-a-machine-learning-classification-model-with-pycaret/loaded-dataset.PNG)
+
+### Training and evaluating our ML classification model
+To train and evaluate our ML model, we need to use the `setup()` function. The function creates our ML transformation pipeline and initializes the environment in PyCaret. PyCaret's rules state that it must be the first function to be called before executing any other function. The `setup()` function takes in two parameters; data, and target. An extra parameter can be added called `categorical_features` and `numeric_features` if you want PyCaret to infer data types in your dataset i.e., infer features with numerical data types into categorical types used in classification. But, we won't use that extra parameter today. I'll introduce it in a follow-up article.
+
+```python
+    exp_name = setup(data = credit_dataset, target='default', session_id=5041)
+```
+
+![Training and evaluating our ML classification model](/engineering-education/building-a-machine-learning-classification-model-with-pycaret/setup-function-results.PNG)
+
+As shown above, running the code generates information concerning the pre-processing pipeline which is constructed when `setup()` is executed. For example, we have 14 `numeric features` and 9 `categorical features` in our data. In our experiment, we've also used the `session_id = 5041` parameter. It's not a must to use it. We've set ours for later reproducibility. Otherwise, leaving it out will prompt a random number to be generated.
```

Comment:
```suggestion
As shown above, running the code generates information concerning the pre-processing pipeline which is constructed when `setup()` is executed. For example, we have 14 `numeric features` and 9 `categorical features` in our data. In our experiment, we've also used the `session_id = 5041` parameter. We've set ours to attain reproducibility. Using it is not compulsory, but excluding it will prompt a random number to be generated.
```